iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_id.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CXX_UNIQUE_ID_INL
+#define IOX_HOOFS_CXX_UNIQUE_ID_INL
+
+#include "iceoryx_hoofs/cxx/newtype.hpp"
+
+#include <atomic>
+
+namespace iox
+{
+namespace cxx
+{
+/// @brief Unique IDs within a process starting with 1. Monotonic increasing IDs are
+/// created with each call to the constructor. The IDs are copy/move constructible/assignable,
+/// comparable, sortable and convertable to the underlying value type.
```

Comment:
I wonder if the name `UniqueId` really fits when the ids are copyable...

Reply:
I was also wondering but came to the conclusion that one must store this IDs in different places. I'm open to better names but we already have a `TypedUniqueId` (this should probably be named differently since it requires a RouDi ID present) class with the same properties

Reply:
I couldn't think of better names. I think it's fine for now but I guess the question will come up again...